peps/pep-0796.rst

```diff
@@ -0,0 +1,264 @@
+PEP: 796
+Title: Relative Virtual Environment Home
+Author: Richard Levasseur <richardlev@gmail.com>
+Sponsor: Alyssa Coghlan <ncoghlan@gmail.com>
+Discussions-To: Pending
+Status: Draft
+Type: Standards Track
+Created: 26-Jun-2025
+Python-Version: 3.15
+
+.. highlight:: rst
+
+
+Abstract
+========
+
+This PEP describes how a relative path for ``home`` in a Python virtual
+environment's ``pyvenv.cfg`` is understood by the Python startup process.
+Specifically, how it is canonicalized into an absolute path later used
+by the runtime. This small detail is a fundamental building block for
+virtual environments to be more portable.
+
+Motivation
+==========
+
+There are two main motivations for allowing relative paths in ``pyvenv.cfg``.
+
+First, it is currently prescribed that the ``home`` value in ``pyvenv.cfg`` be
+an absolute path. The behavior of relative paths is unspecified. While
+techniques exist to work around this for every other sub-part of a virtual
+environment, the one remaining part without a tenable solution is how the
+Python runtime itself finds ``PYTHONHOME``. This is because, currently, the
+startup process requires absolute paths be used for the ``home`` key in
+``pyvenv.cfg``. If a relative path is used, behavior is unspecified (the
+current implementation ends up making it relative to the process's current
+working directory, making it untenable to use).
+
+This requirement is overly proscriptive and restrictive because, given a known
+anchor point, it's easy to transform a relative path to an absolute path and
+still retain predictable and reliable behavior. Thus, the absolute path
+requirement should be relaxed and relative path behavior allowed and defined.
+
+Second, such relative paths are a building block to enable portable virtual
+environments, i.e. copying a virtual environment as-is between hosts of
+compatible platforms. Portable venvs are appealing because virtual environments
+are a popular mechanism for running Python applications. This provides several
+benefits:
+
+* The closer the development environment is to the non-development environment,
+  the more reliable software can be achieved, and the easier it is to reproduce
```

Comment:
I think you may want to rephrase "the more reliable software can be achieved", I find the meaning a bit confusing and it's used both here and in the next bullet?

Reply:
Rephrased a bit. My main point is: the more similar two environments are, the less likely there will be issues due to those environmental differences. Hence its more reliable. Rephrased to more directly say it's about environment-specific differences.